### PR TITLE
Fix using CoilImage with an implicit size

### PR DESCRIPTION
**Pull request by chrisbanes**
Currently `CoilImage` only works when you specific a size Modifier, since it waits for `onPositioned` to be called, and only starts a Coil request when it has a non-zero size. If no Modifier is provided, the `CoilImage` will have a zero size and we never execute a Coil request.

Fixed by moving back to `WithConstraints` to move the request size logic to be based on the incoming constraints, rather than the final size, which is probably more correct anyway.

### CODE REVIEW

**Comment by chrisbanes** [comment]
> Can you create a new constant called `UNSPECIFIED` which can be `Int.MAX_VALUE` behind the scenes? 
Sounds good, done.

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt [hunk 1]
```diff
@@ -224,6 +224,11 @@ fun CoilImage(
     }
 }
 
+/**
+ * Value for a [IntSize] dimension, where the dimension is not specified or is unknown.s
```
Comment (handstandsam) [inline coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:228]: Thanks for making the update!

Minor typo in the javadoc `.s` at the end.

Comment (chrisbanes) [reply coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt]: Woops, thanks!

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt [hunk 2]
```diff
@@ -186,36 +188,50 @@ fun CoilImage(
...
 
-    if (painter == null) {
-        // If we don't have a result painter, we add a Box with our modifier
-        Box(mod) {
-            // If we don't have a result yet, we can show the loading content
-            // (if not null)
-            if (result == null && loading != null) {
-                loading()
+        if (painter == null) {
+            // If we don't have a result painter, we add a Box with our modifier
```
Comment (manuelvicnt) [inline coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:208]: Nit: The modifier is applied to the WithConstraints above now

```suggestion
            // If we don't have a result painter, we add a Box
```

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt [hunk 3]
```diff
@@ -186,36 +188,50 @@ fun CoilImage(
...
     }
 }
 
+/**
+ * Value for a [IntSize] dimension, where the dimension is not specified or is unknown.
+ */
+private const val UNSPECIFIED = -1
+
+@Stable
+private data class MutableRef<T>(var value: T)
```
Comment (manuelvicnt) [inline coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:233]: Smart! Nit: Add some comments about why you need this?

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt (hunk 2 above)
Comment (chrisbanes) [reply coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:208]: Thanks, done in 05f1927

File: coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt (hunk 3 above)
Comment (chrisbanes) [reply coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt:233]: Thanks, done in 05f1927

**Pull request merged**